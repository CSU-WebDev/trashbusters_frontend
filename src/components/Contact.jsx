import React from 'react'
import { Container, Row, Col } from 'react-bootstrap'
import { Button, Form } from 'react-bootstrap'

export default function Contact() {
    return (
        <Container id='contact-container' data-testid="contact-container" className='box'>
            <Row>
                <h3 className="text-center">Contact Us</h3>
            </Row>
            <Row>
                <Col className='d-flex justify-content-center'>
                    <Form id='form' data-testid="form" action="https://formspree.io/f/xyyaglza" method="POST">
                        <Form.Group controlId="formBasicEmail" className="mb-3">
                            <Form.Label className='mb-2'>Email address</Form.Label>
                            <Form.Control type="email" placeholder="Enter email" name="email" required/>
                        </Form.Group>

                        <Form.Group controlId="formBasicName" className="mb-3">
                            <Form.Label className='mb-2'>Name</Form.Label>
                            <Form.Control type="text" placeholder="Enter your name" name="name" required data-testid="name-input"/>
                        </Form.Group>

                        <Form.Group controlId="formBasicMessage" className="mb-3">
                            <Form.Label className='mb-2'>Message</Form.Label>
                            <Form.Control type="textarea" rows="3" placeholder="Enter your message" name="message" required data-testid="message-input"/>
                        </Form.Group>

                        <Button className="btn-lg" variant="primary" type="submit">
                            Submit
                        </Button>
                    </Form>
                </Col>
            </Row>
        </Container>
    )
}
