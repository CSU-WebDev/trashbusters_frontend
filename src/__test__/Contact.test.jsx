import React from 'react';
import { render, screen } from '@testing-library/react';
import Contact from '../components/Contact';
import '@testing-library/jest-dom';

describe('Contact component', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders Contact Us header', () => {
    render(<Contact />);
    expect(screen.getByText(/Contact Us/i)).toBeInTheDocument();
  });

  test('renders form with correct action URL', () => {
    render(<Contact />);
    const form = screen.getByTestId('form');
    expect(form).toHaveAttribute('action', 'https://formspree.io/f/xyyaglza');
  });

  test('renders email input', () => {
    render(<Contact />);
    const emailInput = screen.getByLabelText(/Email address/);
    expect(emailInput).toBeInTheDocument();
    expect(emailInput).toHaveAttribute('type', 'email');
    expect(emailInput).toHaveAttribute('name', 'email');
    expect(emailInput).toHaveAttribute('required', '');
  });

  test('renders name input', () => {
    render(<Contact />);
    const nameInput = screen.getByTestId('name-input');
    expect(nameInput).toBeInTheDocument();
    expect(nameInput).toHaveAttribute('type', 'text');
    expect(nameInput).toHaveAttribute('name', 'name');
    expect(nameInput).toHaveAttribute('required', '');
  });

  test('renders message textarea', () => {
    render(<Contact />);
    const messageInput = screen.getByTestId('message-input');
    expect(messageInput).toBeInTheDocument();
    expect(messageInput).toHaveAttribute('type', 'textarea');
    expect(messageInput).toHaveAttribute('name', 'message');
    expect(messageInput).toHaveAttribute('required', '');
  });

  test('renders submit button', () => {
    render(<Contact />);
    const submitButton = screen.getByRole('button', { name: /Submit/ });
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toHaveAttribute('type', 'submit');
  });
});
