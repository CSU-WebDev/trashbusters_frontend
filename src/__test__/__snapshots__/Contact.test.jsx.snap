// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="box container"
    data-testid="contact-container"
    id="contact-container"
  >
    <div
      class="row"
    >
      <h3
        class="text-center"
      >
        Contact Us
      </h3>
    </div>
    <div
      class="row"
    >
      <div
        class="d-flex justify-content-center col"
      >
        <form
          action="https://formspree.io/f/xyyaglza"
          class=""
          data-testid="form"
          id="form"
          method="POST"
        >
          <div
            class="mb-3"
          >
            <label
              class="mb-2 form-label"
              for="formBasicEmail"
            >
              Email address
            </label>
            <input
              class="form-control"
              id="formBasicEmail"
              name="email"
              placeholder="Enter email"
              required=""
              type="email"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="mb-2 form-label"
              for="formBasicName"
            >
              Name
            </label>
            <input
              class="form-control"
              data-testid="name-input"
              id="formBasicName"
              name="name"
              placeholder="Enter your name"
              required=""
              type="text"
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="mb-2 form-label"
              for="formBasicMessage"
            >
              Message
            </label>
            <input
              class="form-control"
              data-testid="message-input"
              id="formBasicMessage"
              name="message"
              placeholder="Enter your message"
              required=""
              rows="3"
              type="textarea"
            />
          </div>
          <button
            class="btn-lg btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
